lib/publiccloud/provider.pm

```diff
@@ -154,11 +154,14 @@ Creates an ssh keypair in a given file path by $args{ssh_private_key_file}
 =cut
 
 sub create_ssh_key {
-    my ($self, %args) = @_;
-    $args{ssh_private_key_file} //= '/root/.ssh/id_rsa';
-    if (script_run('test -f ' . $args{ssh_private_key_file}) != 0) {
-        assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
-        assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -C "" -m pem -f ' . $args{ssh_private_key_file});
+    my ($self) = @_;
+    my $alg = $self->ssh_key;
+    $alg =~ s@[a-z0-9/-_~.]*id_@@;
+    record_info($alg, "The $alg key will be generated.");
+    my $len = ($alg =~ /rsa/) ? '2048' : '521';
```

Comment:
Why redefine the defaults of `ssh-keygen`? IMHO we could just drop the `-b` part because we're anyways just using the defaults.

Advantage of removing it: When the default key length increases, we don't have to do anything.  
Disadvantage: ???

Reply:
Done, please recheck.